sdk/eventhub/Azure.Messaging.EventHubs/perf/Azure.Messaging.EventHubs.Perf/Azure.Messaging.EventHubs.Perf/Scenarios/Standalone/ConsumeEvents.cs

```diff
@@ -0,0 +1,172 @@
+
+            await Parser.Default.ParseArguments<Options>(args).MapResult(
+                async o => await Run(connectionString, o.Partitions, o.Clients, o.Verbose, o.Send),
+                errors => Task.CompletedTask);
+        }
+
+        public static async Task Run(string connectionString, int partitions, int clients, bool verbose, int send)
+        {
+            if (send > 0)
+            {
+                await SendMessages(connectionString, send);
+            }
+            else
+            {
+                await ReceiveMessages(connectionString, partitions, clients, verbose);
+            }
+        }
+
+        private static async Task ReceiveMessages(string connectionString, int numPartitions, int numClients, bool verbose)
+        {
+            Console.WriteLine($"Receiving messages from {numPartitions} partitions using {numClients} client instances");
+
+            var clients = new EventHubConsumerClient[numClients];
+            for (var i = 0; i < numClients; i++)
+            {
+                clients[i] = new EventHubConsumerClient(connectionString, _eventHubName);
+            }
+
+            try
+            {
+                var client = clients.First();
+                var partitionIds = (await client.GetPartitionIdsAsync()).Take(numPartitions);
+                var partitions = await Task.WhenAll(partitionIds.Select(id => client.GetPartitionPropertiesAsync(id)));
+
+                var totalCount = (long)0;
+                foreach (var partition in partitions)
+                {
+                    var begin = partition.BeginningSequenceNumber;
+                    var end = partition.LastEnqueuedSequenceNumber;
+                    var count = end - begin + 1;
+                    totalCount += count;
+
+                    if (verbose)
+                    {
+                        Console.WriteLine($"Partition: {partition.Id}, Begin: {begin}, End: {end}, Count: {count}");
+                    }
+                }
+                if (verbose)
+                {
+                    Console.WriteLine($"Total Count: {totalCount}");
+                }
+
+                try
+                {
+                    var receiveTasks = new Task<(int messagesReceived, long lastSequenceNumber)>[numPartitions];
+                    var sw = Stopwatch.StartNew();
+                    for (var i = 0; i < numPartitions; i++)
+                    {
+                        receiveTasks[i] = ReceiveAllMessages(clients[i % numClients], partitions[i]);
+                    }
+                    var results = await Task.WhenAll(receiveTasks);
+                    sw.Stop();
+
+                    var elapsed = sw.Elapsed.TotalSeconds;
+                    var messagesReceived = results.Select(r => r.messagesReceived).Sum();
+                    var messagesPerSecond = messagesReceived / elapsed;
+                    var megabytesPerSecond = (messagesPerSecond * _bytesPerMessage) / (1024 * 1024);
```

Comment:
This isn't quite accurate.   There is overhead to a message that is not part of it's payload, particularly on received events where broker properties are populated.    Unfortunately, there's no real accurate way to measure, but we will be underestimating.